tests/acceptance/features/bootstrap/GraphContext.php

```diff
@@ -772,7 +772,7 @@ public function theUserCreatesNewUser(string $user, TableNode $table): void {
 	 */
 	public function addUserToGroup(string $group, string $user, ?string $byUser = null): ResponseInterface {
 		$credentials = $this->getAdminOrUserCredentials($byUser);
-		$groupId = $this->featureContext->getAttributeOfCreatedGroup($group, "id");
+		$groupId = $this->featureContext->getAttributeOfCreatedGroup($group, "id") ?: WebDavHelper::generateUUIDv4();
```

Comment:
I don't think we should generate a fake groupId in case the group doesn't exist in this main function. This generation of fake id is to check for the edge cases of `non-existent` group but in the case of some error the group was not created in the previous `when` or `given` step this will generate a fake id, the test will obviously fail but I think this will make the debugging very confusing? Maybe we can add a filter in the function `$createNonExistentGroup` which defaults to false and only when this filter is true generate the fake id otherwise just follow the normal flow
```suggestion
		$groupId = $this->featureContext->getAttributeOfCreatedGroup($group, "id")
```

Reply:
Since the user addition to the group always has group creation in its `given` statement (not in the `when` step), whether or not a group has been created will be checked inside that test code. The group creation using the `when` step doesn't call this function at all.

Reply:
doesn't matter having adding the fake ID when there is no group is not good logic please add a filter and set it only when we expect the scenario to have no group. Also, why is not the group creation when step not calling this function? how many `addUserToGroup` do we have? we should not have multiple function to do the same task